src/notation/internal/notationinteraction.cpp

```diff
@@ -3880,16 +3880,12 @@ void NotationInteraction::addTextToTopFrame(TextStyleType type)
 
 mu::Ret NotationInteraction::canAddTextToItem(TextStyleType type, const EngravingItem* item) const
 {
-    if (!item) {
-        return false;
-    }
-
     if (isVerticalBoxTextStyle(type)) {
-        return item->isVBox();
+        return item && item->isVBox();
     }
 
     if (type == TextStyleType::FRAME) {
-        return item->isBox();
+        return item && item->isBox() ? make_ok() : make_ret(Err::EmptySelection);
```

Comment:
technically the message is wrong, the selection might not be empty, just not valid for this command - but the only way this can happen is by assigning a shortcut key to the "add frame text" command.